src/main/java/com/example/repick/domain/recommendation/service/RecommendationService.java

```diff
@@ -0,0 +1,160 @@
+            for (int styleIndex : sortedStyleIndices) {
+                if (topRecommendedProducts.size() >= 3) {
+                    break;
+                }
+
+                Category category = Category.fromId(categoryIndex);
+                Style style = Style.fromId(styleIndex);
+
+                List<Product> filteredProducts = productRecommendation.stream()
+                        .filter(product -> product.getProductCategoryList().stream()
+                                .anyMatch(productCategory -> productCategory.getCategory().equals(category)))
+                        .filter(product -> product.getProductStyleList().stream()
+                                .anyMatch(productStyle -> productStyle.getStyle().equals(style)))
+                        .toList();
+
+                topRecommendedProducts.addAll(filteredProducts);
+                topRecommendedProducts = topRecommendedProducts.stream().distinct().limit(3).collect(Collectors.toList());
+            }
+        }
+
+        if (topRecommendedProducts.size() < 3) {
+            topRecommendedProducts = productRecommendation.stream().limit(3).collect(Collectors.toList());
+        }
+
+        return topRecommendedProducts.stream()
+                .map(GetRecommendation::new)
+                .collect(Collectors.toList());
+    }
+
+    public static List<Integer> getSortedIndicesByPreference(List<Double> preferences) {
+        // 인덱스와 조정된 선호도를 저장할 리스트 생성
+        List<int[]> indexedPreferences = new ArrayList<>();
+
+        for (int i = 0; i < preferences.size(); i++) {
+            // 선호도를 랜덤하게 조정
+            double adjustedPreference = preferences.get(i) * Math.random();
+            // 인덱스와 조정된 선호도를 리스트에 추가
+            indexedPreferences.add(new int[]{i, (int)(adjustedPreference * 10000)}); // 정수 변환을 위해 10000을 곱함
+        }
+
+        // 조정된 선호도를 기준으로 리스트 정렬 (내림차순)
+        indexedPreferences.sort((o1, o2) -> {
+            return Integer.compare(o2[1], o1[1]); // 선호도 내림차순 정렬
+        });
+
+        // 정렬된 인덱스를 저장할 리스트 생성
+        List<Integer> sortedIndices = new ArrayList<>();
+        for (int[] pair : indexedPreferences) {
+            sortedIndices.add(pair[0]); // 인덱스만 추출하여 리스트에 추가
+        }
+
+        return sortedIndices;
+    }
+
+
+    public Boolean skipProduct(Long productId) {
+        User user = userRepository.findByProviderId(SecurityContextHolder.getContext().getAuthentication().getName())
+                .orElseThrow(() -> new CustomException(USER_NOT_FOUND));
+
+        skipProduct(user.getId(), productId);
+        Product product = productRepository.findById(productId)
+                .orElseThrow(() -> new CustomException(INVALID_PRODUCT_ID));
+        adjustUserPreferenceOnDetail(user.getId(), product, new double[]{0.7, 0.9, 1.5});
+
+        return true;
+    }
+
+    public void skipProduct(Long userId, Long productId) {
+        userPreferenceProductRepository.save(new UserPreferenceProduct(userId, productId));
+    }
```

Comment:
```suggestion
    public Boolean skipProduct(Long productId) {
        User user = userRepository.findByProviderId(SecurityContextHolder.getContext().getAuthentication().getName())
                .orElseThrow(() -> new CustomException(USER_NOT_FOUND));

        userPreferenceProductRepository.save(new UserPreferenceProduct(userId, productId));
        Product product = productRepository.findById(productId)
                .orElseThrow(() -> new CustomException(INVALID_PRODUCT_ID));
        adjustUserPreferenceOnDetail(user.getId(), product, new double[]{0.7, 0.9, 1.5});

        return true;
    }
```

밑에 있는 skipProduct는 한번 나오기도 하고 이름도 같아서 헷갈리는데 그냥 이렇게 한번에 써도 될 것 같아요

Reply:
이렇게 하면 이미 이전 서비스에서 User를 조회한 상태에서도 한번 더 User를 조회해서 비효율적이지 않을까 하는 생각에 나누어 두었는데요, 그냥 하나로 합치는게 좋을까요?

Reply:
엇 그냥 user.getId()하면 되는거 아니에요?

Reply:
아 그러네요 😅 감사합니당